cita-executor/core/src/contracts/solc/chain_manager.rs

```diff
@@ -35,64 +36,72 @@ lazy_static! {
+    pub fn ext_chain_id(
+        _ext: &mut DataProvider,
+        _gas: &U256,
+        _sender: &Address,
+    ) -> Option<(U256, U256)> {
         trace!("call system contract ChainManagement.ext_chain_id()");
-        let contract = &*CONTRACT_ADDRESS;
+        let _contract = &*CONTRACT_ADDRESS;
         let tx_data = CHAIN_ID_ENCODED.to_vec();
-        let data = &tx_data.as_slice();
-        let mut output = Vec::<u8>::new();
-        match ext.call(
-            gas,
-            sender,
-            contract,
-            None,
-            data,
-            contract,
-            &mut output,
-            CallType::Call,
-        ) {
-            MessageCallResult::Success(gas_left, return_data) => {
-                decode_tools::to_u256(&*return_data).map(|x| (gas_left, x))
-            }
-            MessageCallResult::Reverted(..) | MessageCallResult::Failed => None,
-        }
+        let _data = &tx_data.as_slice();
+        let _output = Vec::<u8>::new();
+        // match ext.call(
+        //     gas,
+        //     sender,
+        //     contract,
+        //     None,
+        //     data,
+        //     contract,
+        //     &mut output,
+        //     CallType::Call,
+        // ) {
+        //     MessageCallResult::Success(gas_left, return_data) => {
+        //         decode_tools::to_u256(&*return_data).map(|x| (gas_left, x))
+        //     }
+        //     MessageCallResult::Reverted(..) | MessageCallResult::Failed => None,
+        // }
+        None
     }
 
     pub fn ext_authorities(
-        ext: &mut Ext,
-        gas: &U256,
-        sender: &Address,
+        _ext: &mut DataProvider,
+        _gas: &U256,
+        _sender: &Address,
         chain_id: U256,
     ) -> Option<(U256, Vec<Address>)> {
         trace!(
             "call system contract ChainManagement.ext_authorities({})",
             chain_id
         );
-        let contract = &*CONTRACT_ADDRESS;
+        let _contract = &*CONTRACT_ADDRESS;
         let mut tx_data = AUTHORITIES_ENCODED.to_vec();
         let param = H256::from(chain_id);
         tx_data.extend(param.to_vec());
-        let data = &tx_data.as_slice();
-        let mut output = Vec::<u8>::new();
-        match ext.call(
-            gas,
-            sender,
-            contract,
-            None,
-            data,
-            contract,
-            &mut output,
-            CallType::Call,
-        ) {
-            MessageCallResult::Success(gas_left, return_data) => {
-                trace!(
-                    "call system contract ChainManagement.ext_authorities() return [{:?}]",
-                    return_data
-                );
-                decode_tools::to_address_vec(&*return_data).map(|addrs| (gas_left, addrs))
-            }
-            MessageCallResult::Reverted(..) | MessageCallResult::Failed => None,
-        }
+        let _data = &tx_data.as_slice();
+        let _output = Vec::<u8>::new();
+        // match ext.call(
```

Comment:
You need to implement this logic.

Reply:
I will.

Reply:
This Logic is similar to eth_call. I will implement the logic of eth_call, and you can reference it.